misc/pause-container/pause.c

```diff
@@ -0,0 +1,57 @@
+// This file is derived from the Kubernetes Project, Copyright 2015 Google Inc. All rights reserved.
+//
+// The original code may be found :
+// https://github.com/kubernetes/kubernetes/blob/master/build/pause/pause.c
+//
+/*
+Copyright 2016 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// Modifications are Copyright 2017 Amazon.com, Inc. or its affiliates. Licensed under the Apache License 2.0
+
+
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+static void sigdown(int signo) {
+  psignal(signo, "Shutting down, got signal");
+  exit(0);
+}
+
+static void sigreap(int signo) {
+  while (waitpid(-1, NULL, WNOHANG) > 0)
+    ;
+}
+
+int main() {
+  if (getpid() != 1)
+    /* Not an error because pause sees use outside of infra containers. */
+    // NOTE: Removal of the word pod from the warning on Line 42
+    fprintf(stderr, "Warning: pause should be the first process\n");
```

Comment:
@thockin Here is the PR against k8s
https://github.com/kubernetes/kubernetes/pull/43578